### PR TITLE
Rip out the port cache

### PR DESCRIPTION
**Pull request by horazont**
For a while now (potentially since ever), ch-k8s-lbaas has been known
for losing floating IPs. We were so far unable to track it down---the
logs were always inconclusive and it wasn't fully clear where the issue
might come from.

As "cache invalidation" is one of the two hard problems in computer
science (next to naming things and off-by-one errors), it seems most
sensible to rip out the port cache.

With the typical cluster size, it should not be necessary to have a
cache. In addition, using a cache is in direct contradiction to acting
on the current state (even though API calls also always return stale
data, but at least it's less stale than your average cache), which we
normally want to do in the Kubernetes world.

So let's try this, for a change.

### CODE REVIEW

**Comment by horazont** [comment]
Test image available at `registry.gitlab.com/jssfr/stash/ch-k8s-lbaas-controller:feature-nocache`. I cannot upload builds here unfortunately, so you'll have to make them yourself.

File: internal/openstack/cache.go [hunk 1]
```diff
@@ -15,150 +15,85 @@
...
 			fips, err := floatingipsv2.ExtractFloatingIPs(page)
 			if err != nil {
 				return false, err
 			}
-			for _, fip := range fips {
-				if fip.PortID == "" {
-					continue
+			for _, found_fip := range fips {
+				if fip != nil {
+					// TODO: warn here?!
```
Comment (sebastianriese) [inline internal/openstack/cache.go:88]: If this isn't supposed to happen warning here would be good (as it may help with debugging).

**Review by sebastianriese — CHANGES_REQUESTED** [review]
Is there a chance to rename `cache.go` something like `ports.go` or would this cause problems?

File: internal/openstack/ports.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ */
+package openstack
+
+import (
+	"github.com/gophercloud/gophercloud"
+	floatingipsv2 "github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/floatingips"
+	portsv2 "github.com/gophercloud/gophercloud/openstack/networking/v2/ports"
+	"github.com/gophercloud/gophercloud/pagination"
+
+  "k8s.io/klog"
```
Comment (sebastianriese) [inline internal/openstack/ports.go:23]: Indentation is messed up here (seems to be tab vs. spaces) – go fmt to fix.

**Comment by horazont** [comment]
Merged as 8179501.

I did not see the comments, but I luckily ran go fmt in the meantime. And then I forgot to force push, so GitHub didn't see the merge correctly.

**Pull request closed without merging**